### PR TITLE
feat(React): React App Analytics (WIP)

### PR DESCRIPTION
**Pull request by jjoyce0510**
**Scope**
These changes only impact the React client. 

**Changes**
This PR introduces a generalized analytics module for tracking:

1. Generic Page View Events
2. Custom Tracking Events (actions, clicks, etc) 

inside the React app. With this PR, we will the ability to easily support to a variety of backend analytics providers like

- Google Analytics
- Mixpanel
- Segment
- Amplitude
- Custom Backend

& many others

We achieve this by building a small extension on top of the popular [Analytics](https://www.npmjs.com/package/analytics) library, which supports many analytics providers out of the box. 

In this PR, we introduce support for Google Analytics, Mixpanel, Amplitude, & a logging provider for testing, as well as implement a series of custom tracking events: 

- LogInEvent
- LogOutEvent
- EntityViewEvent
- SearchEvent
- SearchResultClickEvent
- BrowseResultClickEvent

that will allow you to understand your users behavior on the DataHub platform. 

As of now, configuring an analytics provider (plugin) requires changing code under the "plugins" directory of the analytics module inside `datahub-web-react`. In a follow up PR, we intend to expose these configurations via environment variables on the datahub-frontend-react Docker container, allowing you to configure product analytics without modifying code. 

**Status**
Work in progress. Will be providing a richer README with details on supporting additional providers, configuring a provider, etc. 

**Example**
Mixpanel Dashboard showing metrics collected using the framework.

Screen Shot 2021-04-21 at 12 31 38 PM

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/analytics/event.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+/**
+ * Base Interface for all React analytics events.
+ */
+export interface BaseEvent {
+    type: EventType;
+    actor?: string;
+    timestamp?: number;
+    urn?: string;
+    date?: string;
+    userAgent?: string;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/analytics/event.ts:26]: deviceId --> also remember this is not a session token

File: datahub-web-react/src/app/analytics/event.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    EntitySectionViewEvent,
+    EntityActionEvent,
+}
+
+/**
+ * Base Interface for all React analytics events.
+ */
+export interface BaseEvent {
+    type: EventType;
+    actor?: string;
```
Comment (shirshanka) [inline datahub-web-react/src/app/analytics/event.ts:22]: actorUrn?

File: datahub-web-react/src/app/analytics/event.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    timestamp?: number;
+    urn?: string;
```
Comment (shirshanka) [inline datahub-web-react/src/app/analytics/event.ts:24]: objectUrn?

**Comment by jjoyce0510** [comment]
Taking another pass to address comments & add additional tracking

**Comment by shirshanka** [comment]
Closed via #2499

**Pull request closed without merging**